src/prefect/infrastructure/process.py

```diff
@@ -68,18 +73,22 @@ async def run(
 
         # Open a subprocess to execute the flow run
         self.logger.info(f"Opening process{display_name}...")
-        with tempfile.TemporaryDirectory(suffix="prefect") as tmp_dir:
-            self.logger.debug(
-                f"Process{display_name} running command: {' '.join(self.command)} in {tmp_dir}"
-            )
-
-            process = await run_process(
-                self.command,
-                stream_output=self.stream_output,
-                task_status=task_status,
-                env=self._get_environment_variables(),
-                cwd=tmp_dir,
-            )
+        if self.cwd is None:
+            with tempfile.TemporaryDirectory(suffix="prefect") as tmp_dir:
+                process = await self._create_process(
+                    display_name=display_name,
+                    task_status=task_status,
+                    working_dir=tmp_dir,
+                )
+        else:
+            # We have specified a working directory for the process.
+            # NOTE: This will not clean up the working directory like the TemporaryDirectory will.
+            with tmpchdir(self.cwd) as tmp_dir:
+                process = await self._create_process(
+                    display_name=display_name,
+                    task_status=task_status,
+                    working_dir=tmp_dir,
+                )
```

Comment:
You can use

```python
working_dir_ctx = (
    tempfile.TemporaryDirectory(suffix="prefect")
    if not self.cwd else contextlib.nullcontext(self.cwd)
)

with working_dir_ctx as working_dir:
   ...
```

to avoid having to create branched logic and the separate `_create_process` method.

Reply:
(Also, we do not need to use `tmpchdir` here, we only need it to apply to the subprocess and `tmpchdir` is not async-safe)

Reply:
I love the cleanup here! Thanks for this suggestion!